python/mcap/mcap0/stream_reader.py

```diff
@@ -0,0 +1,133 @@
+import struct
+import zlib
+from io import BufferedReader, BytesIO, RawIOBase
+from typing import Iterator, List, Optional, Tuple, Union
+
+import lz4.frame  # type: ignore
+
+from .data_stream import ReadDataStream
+from .exceptions import McapError
+from .opcode import Opcode
+from .records import (
+    Attachment,
+    AttachmentIndex,
+    ChannelInfo,
+    Chunk,
+    ChunkIndex,
+    DataEnd,
+    Footer,
+    Header,
+    McapRecord,
+    Message,
+    MessageIndex,
+    MetadataIndex,
+    Schema,
+    Statistics,
+    SummaryOffset,
+)
+
+
+def breakup_chunk(chunk: Chunk) -> List[McapRecord]:
+    stream, stream_length = get_chunk_data_stream(chunk)
+    records: List[McapRecord] = []
+    while stream.count < stream_length:
+        opcode = stream.read1()
+        length = stream.read8()
+        if opcode == Opcode.CHANNEL_INFO:
+            channel_info = ChannelInfo.read(stream)
+            records.append(channel_info)
+        elif opcode == Opcode.MESSAGE:
+            message = Message.read(stream, length)
+            records.append(message)
+        elif opcode == Opcode.SCHEMA:
+            schema = Schema.read(stream)
+            records.append(schema)
+        else:
+            # Unknown chunk record type
+            stream.read(length)
```

Comment:
maybe worth exposing some kind of `seek` method to avoid keeping the data in memory?